Minor stylistic improvements to language docs (#725)
hashintel/hash: packages/engine/README.md

> **Warning** - Rust runners are currently not supported. Within your simulation project, you should only see `.js` files within dependencies (for example, dependencies/@hash/age/src/behaviors/age.js). Files ending in `.rs` will be ignored and the run will possibly fail in unclear ways.
>

> > Currently, the easiest way to create a project is by using the [integrated IDE](https://core.hash.ai) called [hcore] (hCore). In the absence of an in-depth description of expected project structure (which will be coming in the future), downloading a project from hCore is currently the easiest way to learn how one should be set out.

Reviewer: ```suggestion
> Currently, the easiest way to create a project is by using [HASH Core](https://core.hash.ai). In the future, an in-depth description of the expected project structure will be given here instead.
```

> In order to download and run a simulation from hCore, use `File -> Export Project` (this is available in the toolbar at the top of the page). For help in finding or creating, and editing, simulations in hCore, take a look at our [online documentation][docs]. Then save and unzip the downloaded project to a location of your choice, for example by

Reviewer: ```suggestion
In order to download and run a simulation from HASH Core, use `File -> Export Project` (this is available in the toolbar at the top of the page). For help in finding or creating, and editing, simulations in HASH Core, take a look at our [online documentation][docs]. Then save and unzip the downloaded project to a location of your choice, for example by
```

> ## Main Concepts

Reviewer 2: This section is actually being superseded by #410 which should have been merged yesterday but I think a few other PRs jumped in front of it and stopped it being in-sync with main :(

#### Stateful

> The [`stateful` crate](./lib/stateful) is responsible for keeping the data in the main simulation loops in sync with the language runners. It encapsulates logic surrounding read/write access, as well as low-level shared memory representation.

Reviewer: Upon review, not actually sure this is correct. We talk about the 'main loop' as a concept (sometimes proper cased in our materials). "main simulation loops" may not necessarily imply the same thing. @Alfred-Mountfield or @TimDiekmann to confirm.

Reviewer 3: `stateful` uses the `memory` crate, which is able to store data in memory. It provides data structures, layouts, and access. It is *not* aware of the simulation loop but is used *by* the simulation loop to keep the data in sync. I see both versions possibly misleading. Could you expand on this section a little bit? The crate-level module describes the `stateful` crate pretty well I think.

Reviewer 3: > handles the sharing of memory between different runners.

This still sounds like the `stateful` crate is actively handling the sharing, but the crate is unaware of the execution crate.

Reviewer 2: Sorry for the wasted time in discussing this, this section is being removed as part of #410 so we don't have to redefine the description of crates in multiple places. The source of truth should be their crate docs.
